### PR TITLE
Fixes #37877 - Make the config file compatible with 0.2.z and 0.4.z

### PR DESCRIPTION
**Pull request by adamruzicka**
- `broker` is an array of mqtt servers used by 0.2
- `server` is an array of mqtt servers used by 0.4
- `data-host` needs to be set to an empty string to override certain compile-time defaults set in RHEL and CentOS Stream
- `prefix` needs to be set to the value we expect to override certain compile-time defaults set in RHEL and CentOS Stream

`broker` and `server` can be set at both once, each will be honored by their respective versions

`data-host` and `prefix` can be set both at once on our build of yggdrasil they'll just duplicate the defaults, on RHEL/CentOS Stream they'll act as overrides

TODOs:
- [x] things in packaging

### CODE REVIEW

**Review by pondrejk — APPROVED** [review]
ack from the functional side, tested together with theforeman/katello-pull-transport-migrate#8

File: app/views/unattended/provisioning_templates/registration/remote_execution_pull_setup.erb [hunk 1]
```diff
@@ -54,34 +54,45 @@ if [ -f $CONFIGTOML ]; then
...
+    # check status of $YGGDRASIL_SERVICE and fail if it is not running
     # possible failure reason: incorrect protocol (should be tcp:// or mqtt://) or port (should be 1883)
     # also, cert-file and key-file must be valid
     # and broker must be running on the server
-    yggdrasil status
+    # yggdrasil-0.2.z provides a dedicated status subcommand, newer yggdrasil doesn't have anything like that
+    if which yggdrasil >/dev/null 2>/dev/null; then
+        yggdrasil status
+    else
+        systemctl status $YGGDRASIL_SERVICE
```
Comment (pondrejk) [inline app/views/unattended/provisioning_templates/registration/remote_execution_pull_setup.erb]: the registration process will stop on this line as the systemctl status will occupy the shell, I think something like `systemctl status $YGGDRASIL_SERVICE --no-pager` should do it

Comment (adamruzicka) [reply app/views/unattended/provisioning_templates/registration/remote_execution_pull_setup.erb]: Or we could just not do it at all? I'd say it is just eyecandy, we still check if the service is active a couple of lines lower

Comment (pondrejk) [reply app/views/unattended/provisioning_templates/registration/remote_execution_pull_setup.erb]: fine with me :)

Comment (adamruzicka) [reply app/views/unattended/provisioning_templates/registration/remote_execution_pull_setup.erb]: Dropped

**Review by ofedoren — APPROVED** [review]
Thanks, @adamruzicka and @pondrejk !

**Pull request merged**